src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -529,6 +529,7 @@ static Linker nativeLinker() {
      * @throws IllegalArgumentException if {@code !address.isNative()}, or if {@code address.equals(MemorySegment.NULL)}.
      * @throws IllegalArgumentException if an invalid combination of linker options is given.
      * @throws IllegalCallerException If the caller is in a module that does not have native access enabled.
+     * @throws OutOfMemoryError if the runtime does not have the memory needed to create the downcall handle.
```

Comment:
Suggestions for the phrasing here are welcome. I think we should use something that works for both downcall handles and upcall stubs though.

Reply:
OOME is pretty much understood to be possible anywhere, given it is a VirtualMachineError. We often do not document it explicitly. The risk with documenting it is that it gives the impression that other methods, which don't document it, can never throw it. A rough grep for `@throws OutOfMemoryError` reveals only 12 classes in java.base that explicitly document this.

Reply:
Taking inspiration from other methods that throw this exception, maybe something like this might work:

`if the downcall method handle cannot be allocated by the Linker`

Reply:
That said, I also agree with @dholmes-ora - e.g. I'm not sure how much OOME is important to document here, since it reflects an internal state of the JVM, rather than something the client can do something about.

E.g. if you create an allocator with `SegmentAllocator::slicingAllocator`, at some point you are going to run out of space in the underlying segment, so it makes sense to report failures (and to document why that happens). But in this case the documentation is going to be very vague, and I don't think it provides a lot of value.

Reply:
Ok. I figured it was similar to Unsafe::allocateMemory, which also documents to OOME. But then again, the user is not directly interested in memory in this case.

I"ll remove these `@throws` tags then